internal/pkg/status/legacy_status.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status
+
+import (
+	"fmt"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	clientu "sigs.k8s.io/cli-experimental/internal/pkg/client/unstructured"
+)
+
+// IsReadyFn - status getter
+type IsReadyFn func(*unstructured.Unstructured) (bool, error)
+
+var legacyTypes = map[string]map[string]IsReadyFn{
```

Comment:
Does the version matter, or is it the same for all API versions for all types?

Reply:
Version might matter. Since v1 is present for almost all objects lets start there. But i will refactor code so it is easy to add versioned behaviour. Alternatively the versioned behaviour could be done in the leaf function. Eg inside podStatus()

Reply:
version may matter. Supporting v1 for now. Adding version support can be done in leaf functions like podReady()